data-processor/src/main/java/io/micronaut/data/processor/visitors/finders/AbstractCriteriaMethodMatch.java

```diff
@@ -673,7 +673,8 @@ protected final boolean hasNoWhereAndJoinDeclaration(@NonNull MethodMatchContext
             return false;
         }
         AnnotationMetadataHierarchy metadataHierarchy = new AnnotationMetadataHierarchy(matchContext.getRepositoryClass(), matchContext.getMethodElement());
-        if (metadataHierarchy.hasAnnotation("io.micronaut.data.jpa.annotation.EntityGraph")) {
+        if (metadataHierarchy.hasAnnotation("io.micronaut.data.jpa.annotation.EntityGraph") ||
```

Comment:
Here and elsewhere in data-processor needed to add handling for that new 'jpa3' package used for hibernate6 module